### PR TITLE
Added new view to integrate Freshdesk SSO

### PR DESCRIPTION
**Pull request by markwooff**
Adds a new page to enable Single Sign-On within Freshdesk (https://github.com/gctools-outilsgc/concierge/issues/45). Although Freshdesk does not integrate directly with OpenID Connect, it can use a HMAC-MD5 hash to confirm accounts from Django using the logic explained here: https://support.freshdesk.com/support/solutions/articles/31166-single-sign-on-remote-authentication-in-freshdesk

### CODE REVIEW

File: core/views.py [hunk 1]
```diff
@@ -208,3 +216,20 @@ def user_sessions_form(request):
...
+    if not request.user:
+        raise Http404()
+
+    name = request.user.name
+    email = request.user.email
+    dt = int(datetime.utcnow().strftime("%s")) - 148
+
+    data = '{0}{1}{2}{3}'.format(name, settings.FRESHDESK_SECRET_KEY, email, dt)
+    generated_hash = hmac.new(settings.FRESHDESK_SECRET_KEY.encode(), data.encode(), hashlib.md5).hexdigest()
+    url = settings.FRESHDESK_URL+'login/sso/?name='+urlquote(name)+'&email='+urlquote(email)+'&'+u'timestamp='+str(dt)+'&hash='+generated_hash+'&debug=1'
```
Comment (bryan-robitaille) [inline core/views.py]: I didn't see the debug parameter in the FreshDesk documentation?  Either way it should probably be removed for prod.  Unless there's a reason to keep it in?

File: core/views.py [hunk 2]
```diff
@@ -21,6 +21,14 @@
 from core.class_views import PleioBackupTokensView, PleioSessionListView, PleioProfileView
 from two_factor.views.profile import DisableView
 
+from django.http import Http404, HttpResponseRedirect
+from django.views.decorators.cache import never_cache
+from django.utils.encoding import iri_to_uri
```
Comment (bryan-robitaille) [inline core/views.py]: Don't see where the iri_uri is used in this snippet?  Left over from dev process?

File: core/views.py [hunk 3]
```diff
@@ -21,6 +21,14 @@
...
+from django.utils.http import urlquote
+from django.core.exceptions import ImproperlyConfigured
```
Comment (bryan-robitaille) [inline core/views.py]: Don't see where the ImproperlyConfigured is used in this code snippet.  Left over?

**Comment by markwooff** [comment]
Thanks @bryan-robitaille - those imports were indeed from past iterations. I've also added the ability to pass in the Freshdesk secret key as an environment variable (easier for deployment).

**Pull request merged**